specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json

```diff
@@ -0,0 +1,2292 @@
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/RulesEngine"
+            }
+          },
+          "default": {
+            "description": "Front Door error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Rules Engine Configuration": {
+            "$ref": "./examples/FrontdoorRulesEngineGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "FrontDoors"
+        ],
+        "description": "Creates a new Rules Engine Configuration with the specified name within the specified Front Door.",
+        "operationId": "RulesEngines_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/frontDoorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/rulesEngineNameParameter"
+          },
+          {
+            "description": "Rules Engine Configuration properties needed to create a new Rules Engine Configuration.",
+            "in": "body",
+            "name": "rulesEngineParameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RulesEngine"
+            }
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/RulesEngine"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new RulesEngine has been created.",
+            "schema": {
+              "$ref": "#/definitions/RulesEngine"
+            }
+          },
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/RulesEngine"
+            }
+          },
+          "default": {
+            "description": "Front Door error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update a specific Rules Engine Configuration": {
+            "$ref": "./examples/FrontdoorRulesEngineCreate.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```

Comment:
> "x-ms-long-running-operation": true [](start = 7, length = 36)

For a Post LRO operation, add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}